lib/ansible/modules/cloud/amazon/cloudfront_facts.py

```diff
@@ -357,33 +478,41 @@ def paginated_response(self, func, result_key=""):
     def keyed_list_helper(self, list_to_key):
         keyed_list = dict()
         for item in list_to_key:
-            aliases = item['Aliases']['Items']
             distribution_id = item['Id']
-            keyed_list.update({distribution_id: item})
-            for alias in aliases:
-                keyed_list.update({alias: item})
+            if 'Items' in item['Aliases']:
+                aliases = item['Aliases']['Items']
+                for alias in aliases:
+                    keyed_list.update( { alias: item } )
+            keyed_list.update( { distribution_id: item } )
         return keyed_list
 
+def set_facts_for_distribution_id_and_alias(details, facts, distribution_id, aliases):
+    facts[distribution_id].update(details)
+    for alias in aliases:
+        facts[alias].update(details)
+    return facts
+
 def main():
     argument_spec = ec2_argument_spec()
     argument_spec.update(dict(
-        distribution_id=dict(required=False, type='str'),
-        invalidation_id=dict(required=False, type='str'),
-        origin_access_identity_id=dict(required=False, type='str'),
```

Comment:
What is the reason for reordering all these parameters? This breaks `blame` attribution for these lines & doesn't improve the code.